src/ranged.cpp

```diff
@@ -1284,6 +1259,40 @@ int time_to_fire(player &p, const itype &firingt)
     return std::max(info.min_time, info.base - info.reduction * p.skillLevel( skill_used ));
 }
 
+static inline void eject_casing( player& p, item& weap ) {
+    itype_id casing_type = weap.get_curammo()->ammo->casing;
+    if( casing_type == "NULL" || casing_type.empty() ) {
+        return;
+    }
+
+    if( weap.has_flag( "RELOAD_EJECT" ) ) {
+        const int num_casings = weap.get_var( "CASINGS", 0 );
+        weap.set_var( "CASINGS", num_casings + 1 );
+        return;
+    }
+
+    item casing( casing_type, calendar::turn, false );
+    casing.charges = 1; // needs charge 1 to stack properly with other casings
+
+    if( weap.has_gunmod( "brass_catcher" ) != -1 ) {
+        p.i_add( casing );
+        return;
+    }
+
+    // Eject casing in random direction avoiding walls using player position as fallback
+    auto brass = closest_tripoints_first( 1, p.pos() );
+    brass.erase( brass.begin() );
```

Comment:
closest tripoint is the place you're standing. I assume he doesn't want to eject at your feet?